lexicon/providers/gandi.py

```diff
@@ -0,0 +1,277 @@
+                if name is not None:
+                    rec['name'] = self._canonicalize_name(name)
+                if content is not None:
+                    rec['value'] = self._txt_encode(content) if rec['type'] == 'TXT' else content
+
+                records = self.api.domain.zone.record.update(self.apikey,
+                                                             self.zone_id,
+                                                             version,
+                                                             {'id': records[0]['id']},
+                                                             rec)
+                if len(records) != 1:
+                    raise GandiInternalError("expected one updated record")
+
+                self.api.domain.zone.version.set(self.apikey, self.zone_id, version)
+                ret = True
+
+            except GandiInternalError:
+                pass
+
+            finally:
+                if not ret and version is not None:
+                    self.api.domain.zone.version.delete(self.apikey, self.zone_id, version)
+
+        print("update_record: {0}".format(ret))
+        return ret
+
+    # Delete an existing record.
+    # If record does not exist, do nothing.
+    # If an identifier is specified, use it, otherwise do a lookup using type, name and content.
+    def delete_record(self, identifier=None, type=None, name=None, content=None):
+        """Removes the specified record in a new Gandi zone."""
+
+        version = None
+        ret = False
+
+        opts = {}
+        if identifier is not None:
+            opts['id'] = identifier
+        else:
+            opts['type'] = type.upper()
+            opts['name'] = self._canonicalize_name(name)
+            opts["value"] = self._txt_encode(content) if opts['type'] == 'TXT' else content
+
+        records = self.api.domain.zone.record.list(self.apikey, self.zone_id, 0, opts)
+        if len(records) == 1:
+            rec = records[0]
+            del rec['id']
+
+            try:
+                version = self.api.domain.zone.version.new(self.apikey, self.zone_id)
+                cnt = self.api.domain.zone.record.delete(self.apikey, self.zone_id, version, rec)
+                if cnt != 1:
+                    raise GandiInternalError("expected one deleted record")
+
+                self.api.domain.zone.version.set(self.apikey, self.zone_id, version)
+                ret = True
+
+            except GandiInternalError:
+                pass
+
+            finally:
+                if not ret and version is not None:
+                    self.api.domain.zone.version.delete(self.apikey, self.zone_id, version)
+
+        print("delete_record: {0}".format(ret))
+        return ret
+
+    def _fqdn(self, name):
+        if not name.endswith('.') and not name.endswith('.{0}'.format(self.domain)):
+            name += '.{0}'.format(self.domain)
+        return name
+
+    def _canonicalize_name(self, name):
```

Comment:
I think this can also be removed and replaced with a call to `_relative_name` from the superclass. https://github.com/AnalogJ/lexicon/blob/master/lexicon/providers/base.py#L98